initdb.d/01_main.sql

```diff
@@ -27,7 +27,9 @@ CREATE TABLE datasets (
     id                  SERIAL PRIMARY KEY,
     stable_id           TEXT UNIQUE,
     title               TEXT,
-    description         TEXT
+    description         TEXT,
+    created_at          TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT clock_timestamp(),
+    valid_until         TIMESTAMP WITH TIME ZONE DEFAULT NULL
```

Comment:
Since there are no validity dates in the messages having this in the DB is kind of pointless.

But to keep in line with the new messages a single entry to show if the dataset has been deprecated will probably have to be added.
```suggestion
    deprecated         BOOL
```

Reply:
i would actually agree with this, because the ga4gh visa issuer should handle the validity of access, + as @jbygdell said not sure where we will get this information about the validity.

Reply:
I strongly disagree tbh. I think it's gonna be extremely useful to know _when_ something got deprecated.

Reply:
Yes but the it would look like this:

```suggestion
    deprecated         TIMESTAMP WITH TIME ZONE DEFAULT NULL
```

Reply:
I can rename the field to `deprecated_at` though if you'd like? :)

Reply:
`valid_until` infers that we know before hand when a dataset will be deprecated.
The current EGA message schema is a reactive one i.e. we get the message at the time when the dataset gets deprecated.

Reply:
I mean - I can remove the field altogether as well, it just seems like functionality that's gonna be needed at some point :D

Reply:
i view the source of information for this being in a different system, and we only need to know if it has been deprecated or not

Reply:
and if there is no actual use case, a bit difficult to understand how would this be used in combination with the information about deprecated datasets from other systems that issue permissions to access the dataset

Reply:
The only use case other that logging is to handle the time gap between a dataset being deprecated and all visas being updated to deny access to the dataset.

Reply:
> One use case other that logging is to handle the time gap between a dataset being deprecated and all visas being updated to deny access to the dataset.

in that case, maybe it would be easier to store unix timestamp in the db as Visas come with unix timestamp.

However if we have these kind of discrepancies between systems that in one system a dataset is deprecated while in the other it is not reflected as deprecated and visas can still be given, then we have a problem: what is the source of truth for that ?

Reply:
I see we have active discussions here 😏 I'm not super clear what will be the use case for a deprecated dataset. Does it mean a user will not be able to access the dataset even when the visa granted the permission? If we can not make an agreement, I suggest we postpone the addition of this field.

Reply:
yeah - I'll remove it for now :) I think it might be useful still, but it's easy to add it later if it's needed!